### ISSUE TITLE
Update the two notebooks to use GeoPandas 0.3 with matplotlib 2

### ISSUE THREAD

**Issue by emiliom** [geohackweek/vector#18]
This issue is a follow up on the problems discussed in #17.

@ocefpaf, can you take the two notebooks @lsetiawan and I created for the Geohackweek vector tutorial (notebooks found [here](https://github.com/geohackweek/tutorial_contents/tree/master/vector/notebooks) and tweak them on a conda env that has matplotlib 2 rather than matplotlib 1.5? The goal is to get nearly the same plot styles but using defaults as much as possible and w/o relying on seaborn styles as done in the notebooks.

You can start looking at the notebooks now (if you're working on Sunday .... or your Monday morning), but there's a complication that will first require help from @lsetiawan. The notebooks use a connection to a PostGIS database hosted on an AWS cloud instance that has access restrictions to University of Washington domains. Hopefully @lsetiawan can open that up on Monday, for a week or so, to let you know work with the notebooks freely.

Thanks! FYI, I'm not just trying to improve these notebooks. The use of matplotlib 2 defaults and styles will help us as we develop materials for the BiGCZ workshop.

**Comment by lsetiawan** [comment]
> AWS cloud instance that has access restrictions to University of Washington domains. Hopefully @lsetiawan can open that up on Monday, for a week or so, to let you know work with the notebooks freely.

The AWS cloud instance for the database should be open to the world. Thanks.

**Comment by emiliom** [comment]
> The AWS cloud instance for the database should be open to the world.

Thanks!! @ocefpaf, let me know if you have any questions on this task.

**Comment by ocefpaf** [comment]
Just got back on the keyboard. Clearing some of the backlog and I'll start with this one soon.

**Comment by emiliom** [comment]
Thanks. You'll find a conda env file for these notebooks at https://github.com/geohackweek/tutorial_contents/tree/master/vector

**Comment by ocefpaf** [comment]
> The AWS cloud instance for the database should be open to the world. Thanks.

@lsetiawan is there something I have to do to get access. I just tried to run the notebooks and crashed at cells [30](http://nbviewer.jupyter.org/github/ocefpaf/tutorial_contents/blob/force_mpl2/vector/notebooks/geopandas_intro.ipynb) and [4](http://nbviewer.jupyter.org/github/ocefpaf/tutorial_contents/blob/force_mpl2/vector/notebooks/geopandas_advanced.ipynb).

**Comment by emiliom** [comment]
@lsetiawan, I guess we need to send him the credential json file? I forget how we did that at the geohackweek. I'll look into it right now.

**Comment by emiliom** [comment]
Nope, it looks like db.json is there, and his notebook read it w/o problems. It really is a connection issue.

**Comment by lsetiawan** [comment]
@ocefpaf Hmm.. I just checked the connection, it seemed fine here. Not sure what's going on. Lemme try running the notebook.

**Comment by emiliom** [comment]
I just tried running the intro notebook, and it worked. I'm in my office at APL, so, within UW network.

**Comment by emiliom** [comment]
@lsetiawan, if you're still puzzled, shoot me an email to tell me what you did to open up access; or we can talk on the phone.

**Comment by lsetiawan** [comment]
@emiliom Anthony and I, never had a chance to restrict connection to only UW Network. It was already open. So I'm not sure what's causing the connection problem for Filipe. I just ran the notebook, and I'm having no trouble.

**Comment by ocefpaf** [comment]
> I just ran the notebook, and I'm having no trouble.

I am at UFBA and things here can be wonky :unamused: 
I'll revert the changes on the notebook and leave only the env changes so you can merge this. Hold on...

**Comment by emiliom** [comment]
Ah, so I misinterpreted what you said this morning: 

> The AWS cloud instance for the database should be open to the world.

I thought you meant you had done something just now to open it up.

Well, I can vouch that it was closed to the world outside the UW even before the Geohackweek event started. I could not do my testing & dev from home w/o being on the APL VPN.

Is Anthony around today? It's possible he did set up this restriction and didn't communicate it well to you. Given that this is the cloud, I can't see how a UW domain restriction would just happen! Also, if need be, I can look at the postgresql access files myself and see if I can find the restriction, and undo it. But you'd have to handhold me with the cloud server shell connection!

@ocefpaf: NO, it's not you! like I said, the connection really was restricted.

**Comment by ocefpaf** [comment]
> NO, it's not you!

Phew! I spend the whole day debugging connection issue, glad to know that at least this one is not me :smile:

**Comment by lsetiawan** [comment]
> Given that this is the cloud, I can't see how a UW domain restriction would just happen! Also, if need be, I can look at the postgresql access files myself and see if I can find the restriction, and undo it. But you'd have to handhold me with the cloud server shell connection!

You'd have to talk to Anthony for shell connection since I don't have the pem key to shell into the database instance unfortunately.

> Is Anthony around today?

He usually is.

> Well, I can vouch that it was closed to the world outside the UW even before the Geohackweek event started. I could not do my testing & dev from home w/o being on the APL VPN.

Hmm... that is really strange. We did not place any restriction since it took too much work, having to call UW IT, etc.

If Anthony is too busy, I can always do a database backup, then deploy the db in our BiGCZ Postgres instance?

**Comment by emiliom** [comment]
>> Well, I can vouch that it was closed to the world outside the UW even before the Geohackweek event started. I could not do my testing & dev from home w/o being on the APL VPN.
>
> Hmm... that is really strange. We did not place any restriction since it took too much work, having to call UW IT, etc.

Yes, I remember you were surprised too when I told you it was clear that the DB was restricted to the UW. (BTW, I assume the test you ran today was while VPN'd; try it from your home network w/o the VPN)

> If Anthony is too busy, I can always do a database backup, then deploy the db in our BiGCZ Postgres instance?

Interesting possibility! Let me try with Anthony first.

**Comment by ocefpaf** [comment]
> I can always do a database backup, then deploy the db in our BiGCZ Postgres instance?

I don't think that is needed. It is probably easier to just run my modifications locally to ensure the PR is OK. (I do know that the right mpl is installed and the graphs that I can create are using the 2.0 style.)

**Comment by lsetiawan** [comment]
> (BTW, I assume the test you ran today was while VPN'd; try it from your home network w/o the VPN)

Nope, I hardly VPN unless I need to access the APL server, since I'm working mostly on BiGCZ stuff, I didn't need to.

**Comment by lsetiawan** [comment]
https://github.com/BiG-CZ/BiG-CZ-Toolbox/tree/master/geopandas/notebooks Doesn't use PostGIS connection.

**Comment by emiliom** [comment]
> Nope, I hardly VPN unless I need to access the APL server, since I'm working mostly on BiGCZ stuff, I didn't need to.

So, you're not on a UW network right now, and you were still able to run the notebook??

**Comment by emiliom** [comment]
> https://github.com/BiG-CZ/BiG-CZ-Toolbox/tree/master/geopandas/notebooks Doesn't use PostGIS connection.

True, but those notebooks are not based on the latest GeoPandas release! They're an offshoot of the Geohackweek 2016 notebooks. ALSO, FYI, I should delete those Maptime materials from that repo ... the final, clean version is actually here, in a stand-alone repo:
https://github.com/emiliom/geopandas-tutorial-maptime

**Comment by lsetiawan** [comment]
> So, you're not on a UW network right now, and you were still able to run the notebook??

Yes. I was able to run the notebook in my home network. 

> True, but those notebooks are not based on the latest GeoPandas release! 

I meant just use the file.

**Comment by emiliom** [comment]
> I meant just use the file.

Good point! Let me look into it, so I can give more useful instructions to @ocefpaf.

**Comment by aaarendt** [comment]
I checked the RDS deployment - it is running fine and still open to everyone. Let me know if there's something else you think I should check.

**Comment by emiliom** [comment]
Thanks for checking, @aaarendt! I guess the problem really is on @ocefpaf's end ... Sigh.

The best course of action is @lsetiawan's recommendation to adapt the PostGIS parts of the notebooks to read files as done on the related notebooks (and file holdings) at https://github.com/emiliom/geopandas-tutorial-maptime. I'll follow up on this later today.

**Comment by ocefpaf** [comment]
> The best course of action is @lsetiawan's recommendation to adapt the PostGIS parts of the notebooks to read files as done on the related notebooks (and file holdings) at https://github.com/emiliom/geopandas-tutorial-maptime. I'll follow up on this later today.

Done in https://github.com/geohackweek/tutorial_contents/pull/18/commits/aa505496c967b21b7365909d3d8747af97fa2107

**Comment by emiliom** [comment]
Thanks, @ocefpaf. I commented on the PR directly, https://github.com/geohackweek/tutorial_contents/pull/18

For reference (specially for me), your fork is here: https://github.com/ocefpaf/tutorial_contents

**Comment by emiliom** [comment]
@ocefpaf, I never followed up on this. Just wanted to say that I did look at your notebooks and the changes in styles with matplotlib 2 a couple of weeks ago. Some changes were a matter of taste, of course, but one in particular had a significant impact (previously distinct colors for different polygons now were no longer distinct). So I decided to leave things the way they are.

Maybe I'll revisit this, but not this year. I'll leave your PR as is (open). Thanks for your work on this. It was helpful, even if I did not end up incorporating it.

Reference to PR's, your fork, or discussions:
https://github.com/geohackweek/tutorial_contents/pull/18
https://github.com/ocefpaf/tutorial_contents/blob/force_mpl2/vector/notebooks/geopandas_intro.ipynb
